From e79020315418aab78bb16aaed5df854b9362b296 Mon Sep 17 00:00:00 2001
From: Andreas Schneider <asn@cryptomilk.org>
Date: Wed, 17 Jan 2018 08:12:31 +0100
Subject: [PATCH 12/14] storage: Allow browseIntent for external storage

This fixes:
  StrictMode: java.lang.Throwable:
    content://com.android.externalstorage.documents/root/75287ed0-b798-42b6-a3e4-a65ad934ea24
    exposed beyond app through Intent.getData() without permission grant
    flags; did you forget FLAG_GRANT_READ_URI_PERMISSION?

Change-Id: I6abb05ed4b8672e777a10c4456b4048cd84ecb35
---
 core/java/android/os/storage/VolumeInfo.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/java/android/os/storage/VolumeInfo.java b/core/java/android/os/storage/VolumeInfo.java
index a21e05e..22d6c02 100644
--- a/core/java/android/os/storage/VolumeInfo.java
+++ b/core/java/android/os/storage/VolumeInfo.java
@@ -438,6 +438,7 @@ public class VolumeInfo implements Parcelable {
         final Intent intent = new Intent(Intent.ACTION_VIEW);
         intent.addCategory(Intent.CATEGORY_DEFAULT);
         intent.setDataAndType(uri, DocumentsContract.Root.MIME_TYPE_ITEM);
+        intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
 
         // note that docsui treats this as *force* show advanced. So sending
         // false permits advanced to be shown based on user preferences.
-- 
2.7.4

